src/interfaces/IAuthorizedEvent.ts

```diff
@@ -3,41 +3,49 @@ export type IAuthorizedEvent<TokenPayload = any> =
   | ILowerCaseAuthorizedEvent<TokenPayload>
   | IUpperCaseAuthorizedEvent<TokenPayload>
 
-export interface IAuthorizedEventBase<TokenPayload = any> {
+export type IAuthorizedEventBase<TokenPayload = any> = {
   /** Authorization information added by this middleware from a JWT. Has to be undefined before hitting the middleware. */
   auth?: {
     payload: TokenPayload
     token: string
   }
   /** An object containing event headers */
   headers: any
-  /** The http request method of this event */
-  httpMethod: string
-}
+} & (
+  | {
+      /** The http request method of this event (for REST APIs) */
+      httpMethod: string
+    }
+  | {
+      /** The metadata about this event (for HTTP APIs) */
+      requestContext: {
+        /** The http request metadata about this event */
+        http: object
+      }
+    }
```

Comment:
Please extract into named interfaces to make this a bit easier to read.

Reply:
I've extracted these into the named interfaces `IAuthorizedRestApiGatewayEventBase` and `IAuthorizedHttpApiGatewayEventBase`